src/seahawk/seahawk_deck/input_xbox_one.py

```diff
@@ -21,164 +21,171 @@
-from std_msgs.msg import Int8MultiArray
-from std_msgs.msg import Float32
-from rclpy.parameter import Parameter
+from std_msgs.msg import Bool
+
 
-class Input(Node):
+class StickyButton():
     """
-    Class that implements the joystick input.
+    Class that implements sticky buttons, meaning a button is pressed to turn it on, 
+    and pressed again to turn it off
     """
 
     def __init__(self):
-        """Initialize this node"""
-        super().__init__('input_xbox_one')
-        """IMPORTANT: The following parameters can only use doubles as their values. Use 0.0 instead of 0 and 1.0 instead of 1."""
-        self.declare_parameter('linear_x_scale', 1.0)  # Forward/Backward
-        self.declare_parameter('linear_y_scale', 1.0)  # Sideways
-        self.declare_parameter('linear_z_scale', 1.0)  # Depth
-        self.declare_parameter('angular_x_scale', 0.0) # Roll (not using roll at the moment)
-        self.declare_parameter('angular_y_scale', 0.5) # Pitch
-        self.declare_parameter('angular_z_scale', 0.5) # Yaw
-        self.subscription = self.create_subscription(Joy, 'joy', self._callback, 10)
-        self.twist_pub = self.create_publisher(Twist, 'drive/twist', 10)
-        self.claw_pub = self.create_publisher(Int8MultiArray, 'claw_control', 10)
-        self.cam_servo_pub = self.create_publisher(Float32, 'camera_control', 10)
-        self.claw_grab = False
-        self.fish_release = False
-        self.bambi_mode = False
-        self.last_a_state = 0
-        self.last_b_state = 0
-        self.last_x_state = 0
-
-    def _callback(self, joy_msg):
-        """Called every time the joystick publishes a message."""
-        self.get_logger().debug(f"Joystick axes: {joy_msg.axes} buttons: {joy_msg.buttons}")
+        """
+        Initialize 'StickyButton' object
+        """
+        self.__feature_state = False    # Is the feature this button controls on (True) or off (False)
+        self.__track_state = 0b0000     # Tracks the last four states of the button using bits
+    
+    def check_state(self, cur_button_state: bool) -> bool:
+        """
+        Checks if a button is toggled on or off and accounts for debouncing
+
+        Args:
+            cur_button_state: Current state of the button provided by the controller
+
+        Returns:
+            True if the button is toggled on, False if off
+        """
+        # Append the current button state to the tracker then removes all but rightmost byte
+        # such that self.__track_state records the most recent four states of the button
+        self.__track_state = (self.__track_state << 1 | cur_button_state) & 0b1111
+
+        # Account for bounce by making sure the last four recorded states
+        # appear to represent a button press. If so, update the feature state
+        if self.__track_state == 0b0011:
+            self.__feature_state = not self.__feature_state
+        return self.__feature_state
+
+    def reset(self):
```

Comment:
No references to this function. Evaluate necessity.

Reply:
Github didn't show any calls to that functionally initially for some reason, it's being used so conversation is no longer relevant.